### PR TITLE
2056-add-id-to-track-token-usage

### PR DESCRIPTION
**Pull request by swordEdge**

### CODE REVIEW

File: src/utils/updateTokenPayloadToSingleToken.ts [hunk 1]
```diff
@@ -3,24 +3,24 @@ import { SingleToken } from '@/types/tokens';
 
 export function updateTokenPayloadToSingleToken(
   payload: UpdateTokenPayload,
+  id?: string,
 ): SingleToken {
   return {
     name: payload.name,
     value: payload.value,
     type: payload.type,
+    $extensions: {
```
Comment (SorsOps) [inline src/utils/updateTokenPayloadToSingleToken.ts:12]: I see we're missing  `...(payload.$extensions ? {` does this cause an issue if we are inadvertently passed a token without an extension property?

File: src/utils/updateTokenPayloadToSingleToken.ts [hunk 2]
```diff
@@ -3,24 +3,24 @@ import { SingleToken } from '@/types/tokens';
 
 export function updateTokenPayloadToSingleToken(
   payload: UpdateTokenPayload,
+  id?: string,
```
Comment (SorsOps) [inline src/utils/updateTokenPayloadToSingleToken.ts:6]: Probably makes more sense to have 
```suggestion
  id: string = uuidv4(),
```
then we don't have to fix the old calls to this function

File: src/app/store/models/tokenState.tsx [hunk 3]
```diff
@@ -130,7 +132,7 @@ export const tokenState = createModel<RootModel>()({
       usedTokenSet: data.usedTokenSets || state.usedTokenSet,
       themes: data.themes || state.themes,
       activeTheme: data.activeTheme || state.activeTheme,
-      tokens: data.sets || state.tokens,
+      tokens: addIdPropertyToTokesn(data.sets ?? {}) || addIdPropertyToTokesn(state.tokens),
```
Comment (SorsOps) [inline src/app/store/models/tokenState.tsx]: The second part will never fire. This should probably be 
```
addIdPropertyToTokesn(data.sets || state.tokens),
```

**Review by SorsOps — CHANGES_REQUESTED** [review]

**Comment by SorsOps** [comment]
Failing tests are primarily because of `storeTokenIdInJsonEditor` now being added

File: src/utils/updateTokenPayloadToSingleToken.ts (hunk 2 above)
Comment (swordEdge) [reply src/utils/updateTokenPayloadToSingleToken.ts:6]: if we should change the id as default parameter, then when we edit the token, tokenId will be changed. That is not what we expect.

File: src/utils/updateTokenPayloadToSingleToken.ts (hunk 1 above)
Comment (swordEdge) [reply src/utils/updateTokenPayloadToSingleToken.ts:12]: About this comment, rightnow, we added the utility to add Id property to all tokens. In the old version, if the payload doesn't have $extension property, we don't add tokenId. But we should add tokenId even though the payload doesn't have $extensions property.

**Review by roppazvan — APPROVED** [review]
LGTM

**Pull request merged**